projects/ADuCM3029_demo_cn0415/src/cn0415.c

```diff
@@ -0,0 +1,2189 @@
+{
+	/* Display verbose menu */
+	usr_uart_write_string(dev->aducm3029_uart_desc, (uint8_t*)"\n");
+	usr_uart_write_string(dev->aducm3029_uart_desc,
+			      (uint8_t*)"Available verbose commands:\n");
+	usr_uart_write_string(dev->aducm3029_uart_desc, (uint8_t*)"\n");
+
+	/* Help command */
+	cn0415_help_general(dev, true);
+
+	/* PWM commands */
+	cn0415_help_pwm(dev, true);
+
+	/* ADC commands */
+	cn0415_help_adc(dev, true);
+
+	/* Dither commands */
+	cn0415_help_dither(dev, true);
+
+	/* PID controller commands */
+	cn0415_help_pid(dev, true);
+
+	/* Application commands */
+	cn0415_help_app(dev, true);
+
+	/* Display short commend menu */
+	usr_uart_write_string(dev->aducm3029_uart_desc, (uint8_t*)"\n");
+	usr_uart_write_string(dev->aducm3029_uart_desc,
+			      (uint8_t*)"Available short commands:\n");
+	usr_uart_write_string(dev->aducm3029_uart_desc, (uint8_t*)"\n");
+
+	/* Help command */
+	cn0415_help_general(dev, false);
+
+	/* PWM commands */
+	cn0415_help_pwm(dev, false);
+
+	/* ADC commands */
+	cn0415_help_adc(dev, false);
+
+	/* Dither commands */
+	cn0415_help_dither(dev, false);
+
+	/* PID controller commands */
+	cn0415_help_pid(dev, false);
+
+	/* Application commands */
+	cn0415_help_app(dev, false);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Echoes characters received from CLI. Implements CLI feedback.
+ *
+ * @param dev  - The device structure.
+ *
+ * @return CN415_SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t cn0415_parse(struct cn0415_dev *dev)
+{
+	int32_t ret;
+	uint8_t c = 0, rdy = 1;
+
+	/* Wait and read a character from UART */
+	ret = usr_uart_read_nb(dev->aducm3029_uart_desc, &c, &rdy);
+	if(ret != 0)
+		return ret;
+
+	/* Decide what to do with it */
+	if(rdy == 1) {
+		switch(c) {
+		case _LF:
+		case _CR:
+			uart_cmd = UART_TRUE;
+			break;
+		case _BS:
+			if(uart_line_index == 0)
+				break;
+			uart_line_index--;
+			break;
+		case _NC:
+			break;
+		default:
+			uart_current_line[uart_line_index++] = c;
+			if(uart_line_index == 256) {
+				uart_line_index--;
+			}
+		}
+
+		uart_current_line[uart_line_index] = '\0';
+	}
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Implements the CLI logic.
+ *
+ * @param dev  - The device structure.
+ *
+ * @return CN415_SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t cn0415_process(struct cn0415_dev *dev)
+{
+	int32_t  ret = 0;
+	cmd_func func = NULL;
+	uint8_t i = 0;
+	int16_t pid_ctrl_adjust;
+	uint16_t command = 0;
+
+	/* Disable timer interrupts to avoid hanging */
+	NVIC_DisableIRQ(TMR1_INT);
+	NVIC_DisableIRQ(TMR2_INT);
+
+	/* Get character and identify it */
+	ret = cn0415_parse(dev);
+	if(ret !
```

Comment:
this is pretty difficult to read. Maybe rename the local variables to make them shorter: `ext_measurement_current -> ext_meas_cur` or something